kotlinx-coroutines-test/src/TestBuilders.kt

```diff
@@ -5,8 +5,88 @@
 package kotlinx.coroutines.test
 
 import kotlinx.coroutines.*
+import kotlinx.coroutines.selects.select
+import java.lang.StringBuilder
 import kotlin.coroutines.*
 
+private const val DEFAULT_TEST_TIMEOUT = 30_000L
+
+/**
+ * A strategy for waiting on coroutines executed on other dispatchers inside a [runBlockingTest].
+ *
+ * Most tests should use [MultiDispatcherWaitConfig]. As an optimization, a test that executes coroutines only on a
+ * [TestCoroutineDispatcher] and never interacts with other dispatchers may use [SingleDispatcherWaitConfig].
+ *
+ * A test may subclass this to customize the wait in advanced cases.
+ */
+interface WaitConfig {
```

Comment:
Agreed - this definitely seems too much to add for a simple parameter that can be addressed with documentation.

Do you have ideas for integrating with an arbitrary dispatcher? I'm not seeing an obvious place to do it but agreed it would be preferred.

Reply:
I agree with Roman, this class seems to be a little bit overkill, integer constant should be enough.

Reply:
+1 to all of the above. After getting this feedback and having more time to think about it, I think this should be a simple timeout.

Reply:
Rewrote this to be a simple timeout.